src/Stratis.Bitcoin.Features.Consensus/ConsensusController.cs

```diff
@@ -45,6 +49,35 @@ public uint256 GetBestBlockHashRPC()
             return this.ChainState.ConsensusTip?.HashBlock;
         }
 
+        /// <summary>
+        /// Get the threshold states of softforks currently being deployed.
+        /// Allowable states are: Defined, Started, LockedIn, Failed, Active.
+        /// </summary>
+        /// <returns>A <see cref="JsonResult"/> object derived from a list of
+        /// <see cref="ThresholdStateModel"/> objects - one per deployment.
+        /// Returns an <see cref="ErrorResult"/> if the method fails.</returns>
+        [Route("api/[controller]/deploymentflags")]
+        [HttpGet]
+        public IActionResult DeploymentFlags()
+        {
+            try
+            {
+                ConsensusRuleEngine ruleEngine = this.ConsensusManager.ConsensusRules.GetRule<SetActivationDeploymentsFullValidationRule>().Parent;
```

Comment:
this is a very inefficient way to get `ConsensusRuleEngine`. 
I think it's on DI so we can get it in the constructor of the controller

Reply:
Done